### PR TITLE
feat(tikv/client-c): enable prow bot for reviewing and merging

### PR DESCRIPTION
**Pull request by wuhuizuo**
Signed-off-by: wuhuizuo <wuhuizuo@126.com>

### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

### CODE REVIEW

File: prow/config/config.yaml [hunk 1]
```diff
@@ -7,20 +7,66 @@ prowjob_namespace: ${PROWJOB_NAMESPACE}
...
+                  - "DCO"
+                  - "idc-jenkins-ci/test"
+                strict: false
+              restrictions: *branch-protection_restrictions
+            release-8.5: *client-c-branch-pt
+            release-8.1: *client-c-branch-pt
+            release-7.5: *client-c-branch-pt
+            release-7.1: *client-c-branch-pt
+            release-6.5: *client-c-branch-pt
+            release-6.1: *client-c-branch-pt
```
Comment (wuhuizuo) [inline prow/config/config.yaml]: /cc @gengliqi
PTAL this part, the `release-8.5` is prepared for future LTS branch.

Comment (wuhuizuo) [reply prow/config/config.yaml]: `strict` is fixed to `true` in the next commit.

**Comment by wuhuizuo** [comment]
/hold
wait for approval from destination repo.

**Review by purelind — APPROVED** [review]
/lgtm

**Review by gengliqi — APPROVED** [review]

**Pull request merged**